src/common/notifications.cpp

```diff
@@ -0,0 +1,89 @@
+#include "pch.h"
+#include "notifications.h"
+
+#include <unknwn.h>
+#include <winrt/base.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Data.Xml.Dom.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.UI.Notifications.h>
+#include <winrt/Windows.ApplicationModel.Background.h>
+
+using namespace winrt::Windows::ApplicationModel::Background;
+using winrt::Windows::Data::Xml::Dom::XmlDocument;
+using winrt::Windows::UI::Notifications::ToastNotification;
+using winrt::Windows::UI::Notifications::ToastNotificationManager;
+
+namespace
+{
+    constexpr std::wstring_view TASK_NAME = L"PowerToysBackgroundNotificationsHandler";
+    constexpr std::wstring_view TASK_ENTRYPOINT = L"PowerToysNotifications.BackgroundHandler";
+    constexpr std::wstring_view APPLICATION_ID = L"PowerToys";
+}
+
+void notifications::register_background_toast_handler()
+{
+    try
+    {
+        // Re-request access to clean up from our previous versions
+        BackgroundExecutionManager::RemoveAccess();
+        BackgroundExecutionManager::RequestAccessAsync().get();
+
+        BackgroundTaskBuilder builder;
+        ToastNotificationActionTrigger trigger{ APPLICATION_ID };
+        builder.SetTrigger(trigger);
+        builder.TaskEntryPoint(TASK_ENTRYPOINT);
+        builder.Name(TASK_NAME);
+
+        const auto tasks = BackgroundTaskRegistration::AllTasks();
+        const bool already_registered = std::any_of(begin(tasks), end(tasks), [=](const auto& task) {
+            return task.Value().Name() == TASK_NAME;
+        });
+        if (already_registered)
+        {
+            return;
+        }
+        (void)builder.Register();
+    }
+    catch (...)
+    {
+        // Couldn't register the background task, nothing we can do
+    }
```

Comment:
@yuyoyuppe.  I think we would want to Log an error here.   It might be helpful if there is a scenario where we don't see people updating to the latest version of powertoys.

Reply:
I agree, let's add telemetry so we can trace problems with the notification.

Reply:
If we are to add these telemetry calls, how are we going to do that? It's a `common` static library, not a DLL/EXE module, so if [define a tracing provider](https://github.com/microsoft/PowerToys/blob/master/src/runner/trace.cpp#L5) like we usually do in other modules, we'll get a "multiple symbols" linking error.
I can come up with a bunch of workarounds for this, but it'll probably require refactoring the way we register providers for all modules.

Reply:
Good point.
For now let's open an issue to track the missing telemetry and the fact that it's not straight forward to add it.
Let's unblock this PR is there is anything else pending.